sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -369,6 +369,23 @@ pi_result _pi_event::start() {
   return result;
 }
 
+bool _pi_event::is_completed() const noexcept {
+  if (!isRecorded_) {
+    return false;
+  }
+  if (!isCompleted_) {
+    const CUresult ret = cuEventQuery(evEnd_);
+    if (ret != CUDA_SUCCESS && ret != CUDA_ERROR_NOT_READY) {
+      PI_CHECK_ERROR(ret);
+      return false;
+    }
+    if (ret == CUDA_ERROR_NOT_READY) {
+      return false;
+    }
```

Comment:
I believe we can safely set `isCompleted_` to `true` here. Will allow it to take a slightly faster path in case this check is made again.

Reply:
Done! I've changed both `_pi_event::is_completed`  and `_pi_event::get_execution_status` to non-`const`.

Reply:
Oh, now I get the comment you made in the summary. It should be fine as non-`const`, but if it goes against your better judgement I am okay with you reverting it and keeping `is_completed` as `const`. I will leave that up to you.

Either way, LGTM! 😄 👍

Reply:
I'm going to go on a limb and say once the status is ` complete` , users aren't likely to query it again, so optimizing this shouldn't be necessary. Reverted ;-).

Reply:
This would leave the "_pi_event" object in a malformed state where the `isCompleted_` is false for a completed event. Light +1 for making this non-const.

Reply:
Another option is to rename it to `hasBeenWaitedOn_` or the like. To my knowledge, `isCompleted_` is only used for this.

Reply:
@smaslov-intel You are right, that is no good either. I'm still feeing a bit iffy about making an `is_*` function non-const though.

I like the idea of `hasBeenWaitedOn_` as well. If there are no objections, I will move forward with this in a couple of hours!

Reply:
Done!